File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    r,l=0,len(board)
    b=[]
    for m in moves:
        for i in range(l):
            if board[i][m-1]:
                if not len(b) or b[-1]!=board[i][m-1]:
                    b+=[board[i][m-1]]
                elif len(b)!=0 or b[-1]==board[i][m-1]:
                    del b[-1]
                    r+=2
                board[i][m-1]=0
                break          
    return r

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int r=0;
    vector<int> b;
    for(int i=0;i<moves.size();i++) {
        for(int j=0;j<board.size();j++) {
            if(board[j][moves[i]-1]>0) {
                if(b.size()==0 or b.back()!=board[j][moves[i]-1])
                    b.push_back(board[j][moves[i]-1]);
                else if(b.back()==board[j][moves[i]-1]) {
                    r+=2;
                    b.pop_back();
                }
                board[j][moves[i]-1]=0;
                break;
            }
        }
    }
    return r;
}